service/src/test/java/org/sunbird/learner/actors/skill/UserSkillManagementActorTest.java

```diff
@@ -315,20 +322,30 @@ private Response createGetUserFailureResponse() {
     return response;
   }
 
-  private void mockGetSkillResponse(String userId) {
+  private void mockGetSkillResponse(String userId, boolean isFailure) {
     HashMap<String, Object> esDtoMap = new HashMap<>();
+
     Map<String, Object> filters = new HashMap<>();
     filters.put(JsonKey.USER_ID, userId);
     esDtoMap.put(JsonKey.FILTERS, filters);
     List<String> fields = new ArrayList<>();
     fields.add(JsonKey.SKILLS);
     esDtoMap.put(JsonKey.FIELDS, fields);
     PowerMockito.mockStatic(ElasticSearchUtil.class);
-    when(ElasticSearchUtil.complexSearch(
-            ElasticSearchUtil.createSearchDTO(esDtoMap),
-            ProjectUtil.EsIndex.sunbird.getIndexName(),
-            ProjectUtil.EsType.user.getTypeName()))
-        .thenReturn(createGetSkillResponse());
+
+    if (!isFailure) {
```

Comment:
restructure the call to avoid code-duplicacy only - response is different rest of the rule is same.